File: tests/e2e/fixtures/bookstore-returning/postgresql/java/src/main/java/com/example/db/com/example/db/queries/Querier.java
package com.example.db.queries;

import com.example.db.models.Author;
import com.example.db.models.Book;
import java.sql.Connection;
import java.sql.SQLException;
import java.util.Optional;
import javax.sql.DataSource;

public final class Querier {
    private final DataSource dataSource;

    public Querier(DataSource dataSource) {
        this.dataSource = dataSource;
    }

    public Optional<Author> createAuthor(String name, String bio, Integer birthYear) throws SQLException {
        try (Connection conn = dataSource.getConnection()) {
            return Queries.createAuthor(conn, name, bio, birthYear);
        }
    }

    public Optional<Author> getAuthor(long id) throws SQLException {
        try (Connection conn = dataSource.getConnection()) {
            return Queries.getAuthor(conn, id);
        }
    }

    public Optional<Author> updateAuthorBio(String bio, long id) throws SQLException {
        try (Connection conn = dataSource.getConnection()) {
            return Queries.updateAuthorBio(conn, bio, id);
        }
    }

    public Optional<Queries.DeleteAuthorRow> deleteAuthor(long id) throws SQLException {
        try (Connection conn = dataSource.getConnection()) {
            return Queries.deleteAuthor(conn, id);
        }
    }

    public Optional<Book> createBook(long authorId, String title, String genre, java.math.BigDecimal price, java.time.LocalDate publishedAt) throws SQLException {
        try (Connection conn = dataSource.getConnection()) {
            return Queries.createBook(conn, authorId, title, genre, price, publishedAt);
        }
    }
}

File: tests/e2e/fixtures/bookstore-returning/postgresql/kotlin/src/main/kotlin/com/example/db/com/example/db/queries/Querier.kt
package com.example.db.queries

import com.example.db.models.Author
import com.example.db.models.Book
import javax.sql.DataSource

class Querier(private val dataSource: DataSource) {

    fun createAuthor(name: String, bio: String?, birthYear: Int?): Author? =
        dataSource.connection.use { conn -> Queries.createAuthor(conn, name, bio, birthYear) }

    fun getAuthor(id: Long): Author? =
        dataSource.connection.use { conn -> Queries.getAuthor(conn, id) }

    fun updateAuthorBio(bio: String?, id: Long): Author? =
        dataSource.connection.use { conn -> Queries.updateAuthorBio(conn, bio, id) }

    fun deleteAuthor(id: Long): Queries.DeleteAuthorRow? =
        dataSource.connection.use { conn -> Queries.deleteAuthor(conn, id) }

    fun createBook(authorId: Long, title: String, genre: String, price: java.math.BigDecimal, publishedAt: java.time.LocalDate?): Book? =
        dataSource.connection.use { conn -> Queries.createBook(conn, authorId, title, genre, price, publishedAt) }
}
